synthesizer/src/vm/finalize.rs

```diff
@@ -1750,4 +1799,589 @@ finalize compute:
+        for private_key in validators.keys() {
+            let address = Address::try_from(private_key).unwrap();
+            let amount = remaining_supply / NUM_VALIDATORS as u64;
+            public_balances.insert(address, amount);
+            allocated_supply += amount;
+        }
+        let address = Address::try_from(validators.keys().next().unwrap()).unwrap();
+        let amount = <CurrentNetwork as Network>::STARTING_SUPPLY - allocated_supply;
+        public_balances.entry(address).and_modify(|balance| *balance += amount).or_insert(amount);
+
+        // Construct the bonded balances.
+        let bonded_balances = validators
+            .iter()
+            .map(|(private_key, (amount, _))| {
+                let address = Address::try_from(private_key).unwrap();
+                (address, (address, *amount))
+            })
+            .chain(delegators.iter().map(|(private_key, (validator, amount))| {
+                let address = Address::try_from(private_key).unwrap();
+                (address, (*validator, *amount))
+            }))
+            .collect();
+
+        println!("[VM2] Generating the genesis block.");
+
+        // Construct the genesis block.
+        let genesis_2 = vm_2
+            .genesis_quorum(validators.keys().next().unwrap(), committee, public_balances, bonded_balances, rng)
+            .unwrap();
+
+        println!("[VM2] Adding the genesis block to the VM.");
+
+        // Add the genesis block to the VM.
+        vm_2.add_next_block(&genesis_2).unwrap();
+
+        println!("Checking that all mappings in `credits.aleo` are equal across the two VMs.");
+
+        // Check that all mappings in `credits.aleo` are equal across the two VMs.
+        let program_id = ProgramID::from_str("credits.aleo").unwrap();
+        let committee_mapping_name = Identifier::from_str("committee").unwrap();
+        let bonded_mapping_name = Identifier::from_str("bonded").unwrap();
+        let unbonding_mapping_name = Identifier::from_str("unbonding").unwrap();
+        let account_mapping_name = Identifier::from_str("account").unwrap();
+
+        let committee_1 = vm_1.finalize_store().get_mapping_confirmed(program_id, committee_mapping_name).unwrap();
+        let committee_2 = vm_2.finalize_store().get_mapping_confirmed(program_id, committee_mapping_name).unwrap();
+        assert_eq!(committee_1, committee_2);
+
+        let bonded_1 = vm_1.finalize_store().get_mapping_confirmed(program_id, bonded_mapping_name).unwrap();
+        let bonded_2 = vm_2.finalize_store().get_mapping_confirmed(program_id, bonded_mapping_name).unwrap();
+        assert_eq!(bonded_1, bonded_2);
+
+        let unbonding_1 = vm_1.finalize_store().get_mapping_confirmed(program_id, unbonding_mapping_name).unwrap();
+        let unbonding_2 = vm_2.finalize_store().get_mapping_confirmed(program_id, unbonding_mapping_name).unwrap();
+        assert_eq!(unbonding_1, unbonding_2);
+
```

Comment:
With #2213 merged, we should check the number of delegators and validators in the `metadata` mapping are equivalent.